src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -107,48 +100,60 @@ export default class PhotoField extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const newView = this.state.src && (prevState.src !== this.state.src || (prevState.done && !this.state.done));
+    const newView = this.state.src && (prevState.src !== this.state.src || ((this.state.progress === 100) && (this.state.progress < 100)));
     const cropper = this.refs.cropper;
     if (cropper && newView) {
       this.setCropBox();
     }
   }
 
   onEdit = () => {
-    this.setState({ done: false, warningMessage: null });
+    this.setState({ progress: 0, warningMessage: null });
   }
 
   onDone = () => {
-    this.setState({ done: true, warningMessage: null });
+    const filePath = 'data.photo';
+    this.props.formContext.uploadFile(
+      this.state.cropResult.file,
+      filePath,
+      this.props.uiSchema['ui:options'],
+      this.updateProgress,
+      false
+    ).catch(() => {
+      // rather not use the promise here, but seems better than trying to pass
+      // a blur function
+      // this.props.onBlur(`${this.props.idSchema.$id}_${idx}`);
```

Comment:
Since we are restricting the file editing parameters, and the file types that can be uploaded for editing, file validation isn't really necessary here. Feedback on how best to communicate relevant errors to users is especially welcome.

Reply:
This could also be rejected because of an network or backend error but neither of these rejections returns an error message. We could refactor this promise to return an error message and display it. e.g. "Check our network settings and try again" or go with something generic. Enabling retrying might require some rework, though. 

https://github.com/department-of-veterans-affairs/vets-website/blob/c73fce410fd64ad4d478cc087fa481a4fe4cdb6f/src/js/common/schemaform/actions.js#L178-L227

Reply:
The normal way we handle errors with file fields is to set the error text in the file object in the form data (see the dispatch on line 204). Any error, networking or otherwise, will result in the component getting new props that include an error message.

Reply:
We have decided to present editing errors ("Zooming in this close will cause your ID photo to look unclear) using a warning style that is distinctive from our default error messaging. Should we make use of that pattern to communicate this kind of error as well?

Reply:
Probably, though you might want to confirm with Lauren and Stacy.